esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -1285,17 +1293,67 @@ bool CWsDfuEx::DFUDeleteFiles(IEspContext &context, IEspDFUArrayActionRequest &r
                     emsg.replaceString("Create ", "Delete ");
 
                 VStringBuffer message("Could not delete file %s: %s.", fn, emsg.str());
-                addResult(df->queryLogicalName(), group, true, message);
+                addResult(dfsLFN.get(), group, true, message);
                 e->Release();
             }
             catch(...)
             {
                 VStringBuffer message("Could not delete file %s: unknown exception.", fn);
-                addResult(df->queryLogicalName(), group, true, message);
+                addResult(dfsLFN.get(), group, true, message);
             }
         }
+        bool checkOwningSuperFile(const char *lfn, const char *group, bool removeFromSuperFiles, bool removeEmptyOwningSuperFiles, StringArray &emptyOwningSuperFiles)
+        {
+            Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(lfn, userdesc, true, false, false, nullptr, defaultPrivilegedUser, 30000); // 30 sec timeout
+            if (!df)
+            {
+                VStringBuffer message("Could not delete file %s: file not found.", lfn);
+                addResult(lfn, group, true, message);
+                return false;
+            }
 
-        void addResult(const char* fileName, const char* nodeGroup, bool failed, const char *message)
+            Owned<IDistributedSuperFileIterator> superOwners = df->getOwningSuperFiles();
+            if (!removeFromSuperFiles && superOwners->first())
```

Comment:
in fact, it's confusing that passes this block if removeFromSuperFiles==false and there are 0 super owners.
Would be clearer if :
```
            if (!removeFromSuperFiles)
            {
                if (!superOwners->first()) // no owners
                    return true;
                VStringBuffer message("The file %s can not be deleted because it is owned by one or more superfiles.", lfn);
                addResult(lfn, group, true, message);
                return false;
            }
```